scala3doc-testcases/src/tests/complexNames.scala

```diff
@@ -0,0 +1,12 @@
+package tests
+
+package complexNames
+
+abstract class A:
+  def ++(other: A): A
+  def +:(other: Int): A
+  def :+(other: Int): A
+  // scala3doc has problems with names in backticks
+  // def `multi word name`: Int
```

Comment:
We probably need a issue for it

cc @nicolasstucki

Reply:
what are we missing here?

Reply:
Print a name that needs backticks in backticks, I believe.

This makes sense to include in Tasty-Reflect, since the general concept is being able to print a name in a form that corresponds to how it was entered in source. I'm not certain what's the best way to do that, though.

Reply:
We will need to introduce a `Name` abstraction in the reflection API. A String loses too much information about the name. We cannot distinguish semantic names from simple names and we cannot know if they were backticked. That is on my TODO list.

Reply:
Probably for now we can just fix that in scala3doc: https://github.com/lampepfl/scala3doc/issues/235#issuecomment-722239956

Reply:
@kordyjan I'd rather avoid the code duplication, but I suppose that the regex for Scala identifiers that don't need a backtick is unlikely to change anytime soon. Can you wrap the code for detecting whether an identifier needs a backtick in a function whose name starts with `hack`, so we can grep for it easily? We have a bunch of functions like that in the doctool already.

Reply:
Actually, we won't have that info. The scaladoc will need to figure it out.

Reply:
@abgruszecki: #10199 is ready.